delete-tags.js

```diff
@@ -0,0 +1,49 @@
+module.exports = async ({ github, context, core }) => {
+  const { IMAGE_NAME, ORGANIZATION } = process.env
+
+  let tagName
+
+  if (context.payload.ref_type === 'branch') {
+    tagName = `branch-${context.payload.ref}`
+  } else {
+    tagName = context.payload.ref.match(/^v(.*)$/)[1]
+  }
+
+  let didDelete = false
+
+  for await (const response of github.paginate.iterator(
+    github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg, {
+      package_type: 'container',
+      package_name: IMAGE_NAME,
+      org: ORGANIZATION
+    })) {
+    for (const version of response.data) {
+      const tags = version.metadata?.container?.tags
+      if (tags?.includes(tagName)) {
+        core.notice(`Package version ${version.html_url} matches tag ${tagName} and will be deleted`)
+
+        const otherTags = tags.filter((tag) => tag !== tagName)
+        if (otherTags.length) {
+          core.warning(`Image version has other tags that will be lost: ${otherTags}`)
+        }
+
+        await github.rest.packages.deletePackageVersionForOrg({
+          package_type: 'container',
+          package_name: IMAGE_NAME,
+          org: ORGANIZATION,
+          package_version_id: version.id
+        })
+
+        didDelete = true
+        break
+      }
+    }
+    if (didDelete) {
```

Comment:
**suggestion (non-blocking)**:

Perhaps silly - you can label a loop and break it explicitly:

```javascript
outer: for (var i of [1,2,3]) {
    for (var j of [4,5,6]) {
        if (i == 2) {
            break outer;
        }
        console.log(`${i}: ${j}`);
    }
}
```

Outputs:

```
1: 4
1: 5
1: 6
```

This means this `if`- block can be removed, and lines 37 and 38 become `break some-outer-loop-label`. Though, that doesn't help line 46.

You could, instead, do the same as `post-url.js` with `return`. I think the logic and behavior is the same?

Reply:
[WHAT IS THIS SORCERY](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label)!? I have never known about this feature, that is _amazing_.